src/topology/algebra/module/weak_dual.lean

```diff
@@ -76,6 +76,17 @@ instance [comm_semiring 𝕜] [add_comm_group E] [module 𝕜 E] [add_comm_monoi
   [module 𝕜 F] (B : E →ₗ[𝕜] F →ₗ[𝕜] 𝕜) : add_comm_group (weak_bilin B) :=
 by { dunfold weak_bilin, apply_instance }
 
+instance module_weak_bilin [comm_semiring 𝕜] [comm_semiring 𝕝] [add_comm_group E] [module 𝕜 E]
+  [add_comm_group F] [module 𝕜 F] [module 𝕝 E] (B : E →ₗ[𝕜] F →ₗ[𝕜] 𝕜) :
+  module 𝕝 (weak_bilin B) :=
+by { dunfold weak_bilin, apply_instance }
```

Comment:
```suggestion
@[priority 100]
instance module_weak_bilin [comm_semiring 𝕜] [comm_semiring 𝕝] [add_comm_group E] [module 𝕜 E]
  [add_comm_group F] [module 𝕜 F] [module 𝕝 E] (B : E →ₗ[𝕜] F →ₗ[𝕜] 𝕜) :
  module 𝕝 (weak_bilin B) :=
by { dunfold weak_bilin, apply_instance }
```
I'm a bit nervous about this instance -- it's not clear to me that Lean won't try to use it to find `module 𝕜 (weak_bilin B)` at some point. I would at the very least lower the priority if not delete it altogether.

Reply:
I thought it should be safe since the instance for the module is derived automatically in the definition. I don't know much about how lean finds instances, so if you rather would like to delete the instance it is fine with me.

Reply:
I changed the priority for now, but leaving this issue open.

Reply:
I'm also not sure whether it's bad. Maybe @eric-wieser would know...?

Reply:
This looks harmless to me. If anything, I'd just remove the `derive module` above, so that it always finds the instance this way.

Reply:
Strictly it would be better to prove this without `dunfold` since that inserts `id` terms into the definition.

Use `show_term` to find the actual instance it's using, then you can replace the proof with that.

Reply:
removing the `derive module` instance causes troubles, it then fails to find that instance automatically.